infra/base-images/base-builder/detect_repo.py

```diff
@@ -29,26 +29,37 @@
 
 
 def main():
-  """Function to get a git repos information based on its commit."""
+  """Function to get a git repo's url and name refenced by OSS-Fuzz Dockerfile.
+
+  Raises:
+    ValueError when a commit or a ref is not provided.
+  """
   parser = argparse.ArgumentParser(
-      description='Finds a specific git repo in an oss-fuzz projects docker file.'
+      description="Finds a specific git repo in an oss-fuzz project's docker file."
   )
   parser.add_argument(
       '--src_dir',
-      help='The location of the oss-fuzz projects source directory',
+      help="The location of the oss-fuzz project's source directory.",
       required=True)
+  parser.add_argument(
+      '--repo_name', help='The name of the git repo.')
   parser.add_argument(
       '--example_commit',
-      help='A commit SHA refrencing the projects main repo',
-      required=True)
+      help='A commit SHA referencing the projects main repo.')
+
   args = parser.parse_args()
+  if not args.repo_name and not args.example_commit:
+    raise ValueError(
+        'Requires either an example commit or a repo name to detect repo.')
```

Comment:
`to detect repo location.` is more descriptive.

Reply:
Done.